pkg/service-manager/handlers.go

```diff
@@ -249,33 +249,51 @@ func (h *Handlers) getServiceInstances(params serviceinstance.GetServiceInstance
 
 func (h *Handlers) deleteServiceInstanceByName(params serviceinstance.DeleteServiceInstanceByNameParams, principal interface{}) middleware.Responder {
 	defer trace.Trace("")()
-	e := entities.ServiceInstance{}
+	b := entities.ServiceBinding{}
+	i := entities.ServiceInstance{}
 
 	var err error
 	opts := entitystore.Options{
 		Filter: entitystore.FilterExists(),
 	}
-	err = h.Store.Get(flags.ServiceManagerFlags.OrgID, params.ServiceInstanceName, opts, &e)
+	// TODO (bjung): We always assume there is an assocated binding... this may not always be true
+	err = h.Store.Get(flags.ServiceManagerFlags.OrgID, params.ServiceInstanceName, opts, &b)
+	if err != nil {
+		return serviceinstance.NewDeleteServiceInstanceByNameNotFound().WithPayload(
+			&models.Error{
+				Code:    http.StatusNotFound,
+				Message: swag.String("binding for service instance not found"),
+			})
+	}
+	err = h.Store.Get(flags.ServiceManagerFlags.OrgID, params.ServiceInstanceName, opts, &i)
 	if err != nil {
 		return serviceinstance.NewDeleteServiceInstanceByNameNotFound().WithPayload(
 			&models.Error{
 				Code:    http.StatusNotFound,
 				Message: swag.String("service instance not found"),
 			})
 	}
-	err = h.Store.Delete(flags.ServiceManagerFlags.OrgID, params.ServiceInstanceName, &entities.ServiceInstance{})
+	err = h.Store.SoftDelete(&b)
```

Comment:
what's the background for this change? Why SoftDelete instead of Delete?

Reply:
Because, I can't actually delete the service instance until the binding is removed.  I get an error if I do.  Therefore, I need the entity marked for delete until I can actually delete it.  Also, this is good practice in general as if the entity is lost (it only exists on the channel), the actual deletion work will not get done.  We'll have just orphaned the resource.